smart_simulation/consumer.py

```diff
@@ -0,0 +1,217 @@
+        )
+    return quantity
+
+
+def multi_day(
+    customer_number: str, days: int, start_date: str = "2020-01-01"
+) -> pd.DataFrame:
+    """
+    Generate consumption data for a customer for a given range of days
+
+    Args:
+        start_date: Start date of time range for data creation
+        customer_number: ID number of customer, which should exist in the customers config file
+        days: Number of days to generate data
+
+    Returns:
+        A Pandas DataFrame of product consumption in units of serving
+    """
+    if not isinstance(customer_number, str):
+        logging.exception("customer_number must be a string.")
+        raise TypeError
+
+    if not isinstance(days, int):
+        logging.exception("days must be an int.")
+        raise TypeError
+
+    customer_config = get_customer(customer_number)
+    dates = pd.date_range(start_date, periods=days, freq="D")
+    servings = np.zeros(days, dtype=int)
+    data = {"servings": servings}
+    consumption_servings = pd.DataFrame(data=data, index=dates)
+
+    for index, day in enumerate(consumption_servings.index):
+        day_of_week = day.dayofweek
+        consumption_servings.at[day, "servings"] = single_day(
+            customer_config, day_of_week
+        )
+    return consumption_servings
+
+
+def get_customer(customer_number):
+    """
+        Get a customer from the customers configuration file specified by customer number
+    Args:
+        customer_number: Customer number in the configuration file
+
+    Returns:
+        Customer configuration namedtuple
+    """
+    if customer_number in customers.customers:
+        return customers.customers[customer_number]
+    else:
+        raise Exception(
+            "Customer: "
+            + customer_number
+            + ", does not exist in the configuration file."
+        )
+
+
+def write_output(data: pd.DataFrame, directory_path: pathlib.Path, file_name: str):
+    """
+    Write a Pandas DataFrame to a csv given a path and file name
+    Args:
+        directory_path: Path to write out the file
+        file_name: Name of the file
+        data: Pandas DataFrame
+    """
+    file_name = file_name + ".csv"
+    if not isinstance(data, pd.DataFrame):
+        logging.exception("data must be a Pandas DataFrame.")
+        raise TypeError
+    data.to_csv(directory_path / file_name)
+
+
+def perfect_scale(
+    data: pd.DataFrame, quantity_to_weight: tuple, stock_weight: float
+) -> pd.DataFrame:
+    """
+    Create 'perfect' scale data from a consumer's consumption behavior
+    Args:
+        quantity_to_weight: Consumer's standard serving size as a Random function and input parameters
+        stock_weight: weight of product
+        data: Consumer behavior data
+
+    Returns: Consumer behavior data with a 'weight' column of scale measurements
+
+    """
+    weights = np.zeros(len(data.index), dtype=float)
+    data["weight"] = weights
+    data.at[0, "weight"] = stock_weight
```

Comment:
The function needs all `weight` values set to 0 and the `weight` at initial time stamp to be equal to `stock_weight`. **Throughout the function, should I be setting single instance values differently?**

Reply:
Yeah, typo on my part. I was specifically saying to make it two lines instead of three, like:
```
data = data.assign(weight=0)
data.at[0, 'weight'] = stock_weight
```

It turns out pandas (pre 1.0) used to have a `set_value()` function where you could have chained the operations together like:
```
data = data.assign(weight=0).set_value('weight', 0, stock_weight)
```
Not sure why it was deprecated and I can't think of a good way of chaining the `.at[]` accessor like that.